src/Tizen.Applications.ComponentBased/Tizen.Applications.ComponentBased.Common/BaseComponent.cs

```diff
@@ -0,0 +1,189 @@
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<LowMemoryEventArgs> LowMemory;
+
+        /// <summary>
+        /// Occurs when the system battery is low.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<LowBatteryEventArgs> LowBattery;
+
+        /// <summary>
+        /// Occurs when the system language is chagned.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<LocaleChangedEventArgs> LocaleChanged;
+
+        /// <summary>
+        /// Occurs when the region format is changed.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<RegionFormatChangedEventArgs> RegionFormatChanged;
+
+        /// <summary>
+        /// Occurs when the device orientation is changed.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<DeviceOrientationEventArgs> DeviceOrientationChanged;
+
+        /// <summary>
+        /// Occurs when the device orientation is changed.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<SuspendedStateEventArgs> SuspendedStateChanged;
+
+        /// <summary>
+        /// A component instance ID.
+        /// It will be created after OnCreate method is invoked.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public string Id { get; private set; }
+
+        /// <summary>
+        /// A component ID
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public string ComponentId { get; private set; }
+
+        /// <summary>
+        /// Parent object
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public ComponentBasedApplication Parent { get; private set; }
+
+        /// <summary>
+        /// Finish current component
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public void Finish()
+        {
+            Interop.CBApplication.ComponentFinish(Handle);
+        }
+
+        internal void Bind(IntPtr handle, string compId, string instanceId, ComponentBasedApplication parent)
+        {
+            Handle = handle;
+            Id = instanceId;
+            ComponentId = compId;
+            Parent = parent;
+        }
+
+        /// <summary>
+        /// Overrides this method if you want to specify a type of this component.
+        /// Default component type is Service type.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public virtual ComponentType ComponentType
+        {
+            get
+            {
+                return ComponentType.Service;
+            }
+        }
+
+        /// <summary>
+        /// Overrides this method if want to handle behavior to restore the previous status.
+        /// </summary>
+        /// <param name="c">Contents</param>
```

Comment:
Need to remarks about life time of Bundle

Reply:
Done